src/libs/ascent/runtimes/ascent_vtkh_data_adapter.cpp

```diff
@@ -1506,28 +1507,122 @@ VTKHDataAdapter::StructuredBlueprintToVTKmDataSet
 
     vtkm::Id3 topo_origin = detail::topo_origin(n_topo);
 
-    if (ndims == 2)
+    if(coords_type == "explicit")
     {
-      vtkm::cont::CellSetStructured<2> cell_set;
-      cell_set.SetPointDimensions(vtkm::make_Vec(x_elems+1,
-                                                 y_elems+1));
-      vtkm::Id2 origin2(topo_origin[0], topo_origin[1]);
-      cell_set.SetGlobalPointIndexStart(origin2);
-      result->SetCellSet(cell_set);
-      neles = x_elems * y_elems;
+      if(ndims == 2)
+      {
+        vtkm::Id x_verts = x_elems + 1;
+        vtkm::Id y_verts = y_elems + 1;
+        neles = x_elems * y_elems;
+        nverts = (x_verts) * (y_verts);
+        
+        std::string ele_shape = "quad";
+        vtkm::UInt8 shape_id;
+        vtkm::IdComponent indices_per;
+        detail::VTKmCellShape(ele_shape, shape_id, indices_per);
+        vtkm::cont::ArrayHandle<vtkm::Id> connectivity;
+        connectivity.Allocate(neles * indices_per);
+        auto conn_portal = connectivity.WritePortal();
+        int offset = 0;
+        // Build Connectivity (Polyhedral cells)
+        for (vtkm::Id i = 0; i < x_elems; ++i) 
```

Comment:
I don't think this is related to Polyhedral?

Reply:
copy paste. thanks for catching!